Library/Homebrew/cmd/update.rb

```diff
@@ -181,8 +181,20 @@ def report
         next unless paths.any? { |p| File.dirname(p) == formula_directory }
 
         case status
-        when "A", "M", "D"
+        when "A", "D"
           map[status.to_sym] << repository.join(src)
+        when "M"
+          file = repository.join(src)
+          begin
+            require "formula_versions"
```

Comment:
By loading `formula_versions`, it will load almost the entire codebase. So if something is accidentally broke, users wouldn't be able to run `brew update` to download the fix. Instead they are required to  run `git` manually. I think we should avoid that.